### PR TITLE
Albums list

### PR DESCRIPTION
**Pull request by YesicaNavaB**
## Summary

Albums list

## HEROKU
https://app-wtraining.herokuapp.com/users/2/albums

TOKENS:
NO ADMINISTRADOR
eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6InRhdG9Ad29sb3guY28iLCJhZG1pbmlzdHJhdG9yIjpmYWxzZX0.avZHqSjQBh-udyneyPLkWy7AmaFap4pl4g2Z04s2bqM

ADMINISTRADOR

eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6Im1hcnRhQHdvbG94LmNvIiwiYWRtaW5pc3RyYXRvciI6dHJ1ZX0.KItJN_pcWDR3Zy-IBny9fmlPE1eST1c1TyJ2QVp0If4

## Trello Card

https://trello.com/c/I8OJn6zJ

### CODE REVIEW

File: test/albums.test.js [hunk 1]
```diff
@@ -3,6 +3,19 @@ const request = require('supertest'),
   { factoryCreate, factoryCreateAlbums } = require('../test/utils.test'),
   dictum = require('dictum.js');
 
+const token =
```
Comment (jersonlopez) [inline test/albums.test.js]: move theese object to util.js file

File: test/albums.test.js [hunk 2]
```diff
@@ -39,18 +49,76 @@ describe('album purchase', () => {
...
+      .then(() => {
+        factoryCreateAlbums({ name: 'eaque aut omnis a' });
+      })
+      .then(() =>
+        request(app)
+          .post('/users/sessions')
+          .send({ email: 'yesica@wolox.co', password: 'shdfgs345' })
+          .set('Accept', 'application/json')
+          .then(response => {
+            const tokenResponse = JSON.parse(response.text);
```
Comment (jersonlopez) [inline test/albums.test.js]: if you put only `const tokenResponse = response;`, don't work?

File: test/albums.test.js [hunk 3]
```diff
@@ -39,18 +49,76 @@ describe('album purchase', () => {
...
+            request(app)
+              .get('/users/1/albums')
+              .set({ Accept: 'application/json', Authorization: tokenResponse.token })
```
Comment (jersonlopez) [inline test/albums.test.js]: i think that you can sent `response.token`

File: app/services/albums.js [hunk 4]
```diff
@@ -36,6 +36,13 @@ const albumPurchased = (albumId, userId) =>
       throw error.databaseError(err.message);
     });
 
+exports.getAlbumsListByIdUser = userId =>
+  Album.findAll({ where: { userId } })
+    .then(result => result)
```
Comment (jersonlopez) [inline app/services/albums.js]: I'm not going to say anything here, you know... 😄😄😄😄😄😄

File: app/services/users.js [hunk 5]
```diff
@@ -88,3 +89,20 @@ exports.createUserAdmin = userData => {
   userData.administrator = true;
   return upsert(userData);
 };
+
+exports.userAlbumsList = async req => {
+  try {
+    if (req.body.decode.administrator !== true) {
+      const user = await getUser(req.body.decode.email);
+      if (Number(user.id) !== Number(req.params.user_id)) {
```
Comment (jersonlopez) [inline app/services/users.js]: you can just compare `user.id !== req.params.user_id`. try it and tell me what happen please.

File: app/middlewares/errors.js [hunk 6]
```diff
@@ -12,7 +12,8 @@ const statusCodes = {
   [errors.DEFAULT_ERROR]: 500,
   [errors.VALIDATE_TOKEN_ERROR]: 401,
   [errors.CREATE_ADMIN_ERROR]: 400,
-  [errors.BUY_ALBUMS_ERROR]: 400
+  [errors.BUY_ALBUMS_ERROR]: 400,
+  [errors.USER_ALBUMS_LIST_ERROR]: 401
```
Comment (jersonlopez) [inline app/middlewares/errors.js]: this code is perfect if the user is requiring the album of another user, but if the album is no purchase yet?

File: app/services/users.js [hunk 7]
```diff
@@ -88,3 +89,20 @@ exports.createUserAdmin = userData => {
...
+        throw error.userAlbumsListError('you can only see your albums');
+      }
+    }
+    const getAlbumsList = await servicesAlbums.getAlbumsListByIdUser(req.params.user_id);
```
Comment (jersonlopez) [inline app/services/users.js]: i'm not sure of this but i think that you can do the follow: 
`return await servicesAlbums.getAlbumsListByIdUser(req.params.user_id);`

File: app/services/users.js [hunk 8]
```diff
@@ -88,3 +89,20 @@ exports.createUserAdmin = userData => {
...
+    return getAlbumsList;
+  } catch (err) {
+    throw error.userAlbumsListError(err);
```
Comment (jersonlopez) [inline app/services/users.js]: when you `throw error.userAlbumsListError(err);`, this line is overwriting the `throw` in line 98.

File: app/services/users.js [hunk 9]
```diff
@@ -17,7 +18,7 @@ const upsert = userData =>
       throw error.databaseError(err.message);
     });
 
-exports.getUser = email =>
+const getUser = email =>
```
Comment (jersonlopez) [inline app/services/users.js]: this is fixing in pr #12

**Review by jersonlopez — CHANGES_REQUESTED** [review]

File: app/services/users.js (hunk 5 above)
Comment (YesicaNavaB) [reply app/services/users.js]: if I do it that way I can't make the comparison, so I turn it into numbers so I can compare.

File: test/albums.test.js (hunk 1 above)
Comment (YesicaNavaB) [reply test/albums.test.js]: ok!

File: test/albums.test.js (hunk 2 above)
Comment (YesicaNavaB) [reply test/albums.test.js]: yes!

File: test/albums.test.js (hunk 3 above)
Comment (YesicaNavaB) [reply test/albums.test.js]: yes!

File: app/services/albums.js (hunk 4 above)
Comment (YesicaNavaB) [reply app/services/albums.js]: perfect!

File: app/middlewares/errors.js (hunk 6 above)
Comment (YesicaNavaB) [reply app/middlewares/errors.js]: ok

File: app/services/users.js (hunk 7 above)
Comment (YesicaNavaB) [reply app/services/users.js]: ok!

**Review by jersonlopez — APPROVED** [review]
good job!!!! 👍 💯

**Pull request merged**